libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -456,6 +486,72 @@ export abstract class CommandLineParameterProvider {
       throw new Error('Command Line Parser Data was already processed');
     }
 
+    // Search for any ambiguous parameters and throw an error if any are found
+    for (const [parserKey, parameterName] of this._ambiguousParameterNamesByParserKey) {
+      if (data[parserKey]) {
+        // Determine if the ambiguous parameter is a short name or a long name, since the process of finding
+        // the non-ambiguous name is different for each.
+        const duplicateShortNameParameters: CommandLineParameter[] | undefined =
+          this._parametersByShortName.get(parameterName);
+        if (duplicateShortNameParameters) {
+          // We also need to make sure we get the non-ambiguous long name for the parameter, since it is
+          // possible for that the long name is ambiguous as well.
+          const nonAmbiguousLongNames: string[] = [];
+          for (const parameter of duplicateShortNameParameters) {
+            const matchingLongNameParameters: CommandLineParameter[] | undefined =
+              this._parametersByLongName.get(parameter.longName);
+            if (!matchingLongNameParameters?.length) {
+              // This should never happen
+              throw new Error(
```

Comment:
If this should never happen,  use `InternalError` isntead of `Error`

Reply:
We don't include `InternalError` as a dependency, presumably for bundle-size reasons. That's why I left it this way. I can change if you think it's particularly important.